modules/pubsub/src/main/java/com/spotify/ffwd/pubsub/PubsubPluginSink.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2013-2018 Spotify AB. All rights reserved.
+ *
+ * The contents of this file are licensed under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.spotify.ffwd.pubsub;
+
+import com.google.cloud.pubsub.v1.Publisher;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.Inject;
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.PubsubMessage;
+import com.spotify.ffwd.model.Batch;
+import com.spotify.ffwd.model.Batch.Point;
+import com.spotify.ffwd.model.Event;
+import com.spotify.ffwd.model.Metric;
+import com.spotify.ffwd.output.BatchablePluginSink;
+import com.spotify.ffwd.serializer.Serializer;
+import eu.toolchain.async.AsyncFramework;
+import eu.toolchain.async.AsyncFuture;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class PubsubPluginSink implements BatchablePluginSink {
+
+  @Inject
+  AsyncFramework async;
+
+  @Inject
+  Publisher publisher;
+
+  @Inject
+  private Serializer serializer;
+
+  @Override
+  public boolean isReady() {
+    return true;
+  }
+
+  @Override
+  public void init() {
+
+  }
+
+  @Override
+  public AsyncFuture<Void> sendEvents(Collection<Event> events) {
+    for (Event event : events) {
+      try {
+        publisher.publish(PubsubMessage.newBuilder()
+            .setData(ByteString.copyFrom(serializer.serialize(event))).build()
+        );
+      } catch (Exception e) {
+        log.error("Failed to publish event {}", e);
+      }
+    }
+
+    return async.resolved();
+  }
+
+  @Override
+  public AsyncFuture<Void> sendMetrics(Collection<Metric> metrics) {
+    for (Metric metric : metrics) {
+      try {
+        final PubsubMessage message = PubsubMessage.newBuilder()
+            .setData(ByteString.copyFrom(serializer.serialize(metric)))
+            .build();
+        publisher.publish(message);
+      } catch (Exception e) {
+        log.error("Failed to publish metric {}", e);
+      }
+    }
+
+    return async.resolved();
+  }
+
+  @Override
+  public AsyncFuture<Void> sendBatches(Collection<Batch> batches) {
+    // TODO(dmichel): should this support events?
+    final ArrayList<Metric> metrics = new ArrayList<>();
+
+    batches.forEach(batch -> batch.getPoints().forEach(
+        point -> metrics.add(convertBatchMetric(batch, point))));
```

Comment:
Since this is a nested `forEach`, I think it would be more readable  formatted as
```
batch -> {
   batch.getPoints().forEach(point -> {
      metrics.add(convertBatchMetric(batch, point));
   }
}
```

Reply:
fixed formatting 👍